src/packaging/metadata.py

```diff
@@ -22,9 +22,7 @@
 T = typing.TypeVar("T")
 
 
-if "ExceptionGroup" in builtins.__dict__:  # pragma: no cover
-    ExceptionGroup = ExceptionGroup
-else:  # pragma: no cover
+if "ExceptionGroup" not in builtins.__dict__:  # pragma: no cover
```

Comment:
I flipped the order here to get the second type to stick.

IMO, this should be `if sys.version_info < (3, 11)` instead.

Reply:
Actually, getting the type check right is tricky without it, so switching it to the version check now.